index/field/doc_id_multi_field_intersect_iterator.go

```diff
@@ -0,0 +1,99 @@
+package field
+
+import (
+	"github.com/xichen2020/eventdb/document/field"
+	"github.com/xichen2020/eventdb/index"
+)
+
+// DocIDMultiFieldIntersectIterator is an iterator that iterates over multiple
+// fields for values associated with documents in both the given doc ID set
+// iterator and the multi field iterator.
+type DocIDMultiFieldIntersectIterator struct {
+	docIt        index.DocIDSetIterator
+	multiFieldIt *MultiFieldIterator
+
+	done       bool
+	err        error
+	currDocIDs [2]int32 // 0 for docIt, 1 for multiFieldIt
+	currVals   []field.ValueUnion
+}
+
+// NewDocIDMultiFieldIntersectIterator creates a new DocIDMultiFieldIntersectIterator.
+func NewDocIDMultiFieldIntersectIterator(
+	docIt index.DocIDSetIterator,
```

Comment:
Is this `docIt` iterator the union of the Doc ID sets of the filtered values?

If it is, is the idea for the `DocIDMultiFieldIntersectIterator` to be used to produce the order/sort by fields?

Reply:
Yes and yes.